manticore/platforms/evm.py

```diff
@@ -0,0 +1,1760 @@
+        #if self.pc >= len(self.bytecode):
+        #    return InvalidOpcode('Code out of range')
+        #if self.pc in self.invalid:
+        #    raise InvalidOpcode('Opcode inside a PUSH immediate')
+
+        def getcode():
+            for byte in self.bytecode[self.pc:]:
+                yield byte
+            while True:
+                yield '\x00'
+
+        return EVMDecoder.decode_one(getcode())
+
+    #auxiliar funcs
+    #Stack related
+    def _push(self, value):
+        '''
+                   ITEM0
+                   ITEM1
+                   ITEM2
+             sp->  {empty}
+        '''
+        assert isinstance(value,(int,long)) or isinstance(value, BitVec) and value.size==256 
+        if len(self.stack) >= 1024:
+            raise StackOverflow()
+        self.stack.append(value & TT256M1)
+
+    def _pop(self):
+        if len(self.stack) == 0:
+            raise StackUnderflow()
+        return self.stack.pop()
+        
+
+    #Execute an instruction from current pc
+    def execute(self):
+        if issymbolic(self.pc):
+            expression = self.pc
+            def setstate(state, value):
+                state.platform.current.pc = value
+            raise Concretize("Concretice PC",
+                                expression=expression, 
+                                setstate=setstate,
+                                policy='ALL')
+
+        self.publish('will_decode_instruction', self.pc)
+        current = self.instruction
+
+        self.publish('will_execute_instruction', current)
+
+        implementation = getattr(self, current.semantics, None)
+        if implementation is None:
+            raise TerminateState("Instruction not implemented %s"%current.semantics, testcase=True)
+
+
+
+        #Get arguments (imm, pop)
+        arguments = []
+        if self.instruction.has_operand:
+            arguments.append(current.operand)
+        for _ in range(current.pops):
+            arguments.append(self._pop())
+
+        self.publish('did_execute_instruction', current)
+
+        #Execute
+        try:
+            result = implementation(*arguments)
+        except ConcretizeStack as ex:
+            for arg in reversed(arguments):
+                self._push(arg)
+            def setstate(state, value):
+                self.stack[-ex.pos] = value
+            raise Concretize("Concretice Stack Variable",
+                                expression=ex.expression, 
+                                setstate=setstate,
+                                policy='ALL')
+        except EVMException as e:
+            self.last_exception = e
+            raise
+
+        self.publish('did_execute_instruction', self, current)
+
+        
+        #Check result (push)
+        if current.pushes > 1:
+            assert len(result) == current.pushes
+            for value in reversed(result):
+                self._push(value)
+        elif current.pushes == 1:
```

Comment:
Why does this need a special-cased conditional?

Reply:
With single PUSH instruction handlers wont return a list with 1 element but the element itself.